File: app/src/main/java/com/mobitill/barandrestaurant/data/order/OrderRepository.java
package com.mobitill.barandrestaurant.data.order;

import android.support.annotation.NonNull;
import android.util.Log;

import com.mobitill.barandrestaurant.data.Local;
import com.mobitill.barandrestaurant.data.Remote;
import com.mobitill.barandrestaurant.data.order.model.Order;
import com.mobitill.barandrestaurant.data.order.source.local.OrderDataSource;
import com.mobitill.barandrestaurant.utils.schedulers.BaseScheduleProvider;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;

import static com.google.common.base.Preconditions.checkNotNull;

/**
 * Created by james on 5/24/2017.
 */

public class OrderRepository implements OrderDataSource {

    private static final String TAG = OrderRepository.class.getSimpleName();

    private final OrderDataSource mOrderLocalDataSource;
    private final OrderDataSource mOrderRemoteDataSource;

    @NonNull private final BaseScheduleProvider mSchedulerProvider;

    @Inject
    OrderRepository (
            @Local OrderDataSource orderLocalDataSource,
            @Remote OrderDataSource orderRemoteDataSource,
            @NonNull BaseScheduleProvider scheduleProvider
    ){
        checkNotNull(scheduleProvider, "scheduleProvider should not be null");
        mSchedulerProvider = scheduleProvider;
        mOrderLocalDataSource = orderLocalDataSource;
        mOrderRemoteDataSource = orderRemoteDataSource;
    }

    @Override
    public Observable<List<Order>> getAll() {
        return mOrderLocalDataSource
                .getAll()
                .observeOn(mSchedulerProvider.ui());
    }

    @Override
    public Observable<Order> getOne(String id) {
        Log.i(TAG, "getOne: ");
        return mOrderLocalDataSource
                .getOne(id)
                .observeOn(mSchedulerProvider.ui());
    }

    @Override
    public Order save(Order item) {
        return mOrderLocalDataSource.save(item);
    }

    @Override
    public int delete(String id) {
        return mOrderLocalDataSource.delete(id);
    }

    @Override
    public int update(Order item) {
        return mOrderLocalDataSource.update(item);
    }

    @Override
    public void deleteAll() {
        mOrderLocalDataSource.deleteAll();
    }

    @Override
    public Order getLastCreated() {
        return null;
    }


    @Override
    public Order getOrderFromRowId(Long rowId) {
        return null;
    }

    @Override
    public Observable<List<Order>> getOrdersWithSynced(Integer isSynced) {
        return mOrderLocalDataSource
                .getOrdersWithSynced(isSynced);
    }

    @Override
    public Observable<List<Order>> getOrdersForCheckout(Integer checkout, Integer checkoutFlagged) {
        return mOrderLocalDataSource.getOrdersForCheckout(checkout, checkoutFlagged);
    }
}

File: app/src/main/java/com/mobitill/barandrestaurant/data/order/OrderRepository.kt
package com.mobitill.barandrestaurant.data.order

import android.support.annotation.NonNull
import com.mobitill.barandrestaurant.data.Local
import com.mobitill.barandrestaurant.data.Remote
import com.mobitill.barandrestaurant.data.order.model.Order
import com.mobitill.barandrestaurant.data.order.source.local.OrderDataSource
import com.mobitill.barandrestaurant.data.order.source.local.OrderLocalDataSource
import com.mobitill.barandrestaurant.data.order.source.remote.OrderRemoteDataSource
import com.mobitill.barandrestaurant.utils.schedulers.BaseScheduleProvider
import io.reactivex.Observable
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by andronicus on 5/16/2017.
 */
@Singleton
class OrderRepository @Inject constructor(@Local private val mOrderLocalDataSource: OrderLocalDataSource,
                                          @Remote private val mOrderRemoteDataSource: OrderRemoteDataSource,
                                          @NonNull private val mSchedulerProvider: BaseScheduleProvider) : OrderDataSource{




    override fun getAll(): Observable<MutableList<Order>> {
        return mOrderLocalDataSource
                .all
                .observeOn(mSchedulerProvider.ui())
    }

    override fun getOne(id: String?): Observable<Order> {
        return mOrderLocalDataSource
                .getOne(id)
                .observeOn(mSchedulerProvider.ui())
    }

    override fun save(item: Order?): Long {
        return mOrderLocalDataSource.save(item)
    }

    override fun delete(id: String?): Int {
        return mOrderLocalDataSource.delete(id)
    }

    override fun update(item: Order?): Int {
        return mOrderLocalDataSource.update(item)
    }

    override fun deleteAll() {
        mOrderLocalDataSource.deleteAll()
    }

}
